internal/metarepos/raft_metadata_repository.go

```diff
@@ -1422,7 +1425,7 @@ func (mr *RaftMetadataRepository) registerEndpoint(ctx context.Context) {
 
 func (mr *RaftMetadataRepository) GetClusterInfo(context.Context, types.ClusterID) (*mrpb.ClusterInfo, error) {
 	if !mr.IsMember() {
-		return nil, verrors.ErrNotMember
+		return nil, status.Errorf(codes.PermissionDenied, "not member")
```

Comment:
I am not sure it is PermissionDenied is a good fit. Usually, PermissionDenied indicates that the caller does not have enough permission.

https://github.com/grpc/grpc-go/blob/6a707eb1bb94d73e8a692961da19bb5c95b7b723/codes/codes.go#L81-L86

```go
	// PermissionDenied indicates the caller does not have permission to
	// execute the specified operation. It must not be used for rejections
	// caused by exhausting some resource (use ResourceExhausted
	// instead for those errors). It must not be
	// used if the caller cannot be identified (use Unauthenticated
	// instead for those errors).
```

The client that receives PermissionDenied after calling GetClusterInfo can misunderstand that it has less permission to invoke the RPC.

Reply:
You're right. How about `Unavailable`?

Reply:
If this node does not belong to a metadata repository group yet, but it will soon, Unavailable seems to be good.

https://github.com/grpc/grpc-go/blob/6a707eb1bb94d73e8a692961da19bb5c95b7b723/codes/codes.go#L170-L172
```go
	// Unavailable indicates the service is currently unavailable.
	// This is a most likely a transient condition and may be corrected
	// by retrying with a backoff. Note that it is not always safe to retry
```

You can find differences between Unavailable and FailedPrecondition: 

https://github.com/grpc/grpc-go/blob/6a707eb1bb94d73e8a692961da19bb5c95b7b723/codes/codes.go#L105-L114
```go
	// A litmus test that may help a service implementor in deciding
	// between FailedPrecondition, Aborted, and Unavailable:
	//  (a) Use Unavailable if the client can retry just the failing call.
	//  (b) Use Aborted if the client should retry at a higher-level
	//      (e.g., restarting a read-modify-write sequence).
	//  (c) Use FailedPrecondition if the client should not retry until
	//      the system state has been explicitly fixed. E.g., if an "rmdir"
	//      fails because the directory is non-empty, FailedPrecondition
	//      should be returned since the client should not retry unless
	//      they have first fixed up the directory by deleting files from it.
```